test/orchestratord/mzcompose.py

```diff
@@ -718,6 +713,90 @@ def check_replicas():
         retry(check_replicas, 120)
 
 
+class SystemParamConfigMap(Modification):
+    @classmethod
+    def values(cls) -> list[Any]:
+        return [{"max_connections": 1000}]
+
+    @classmethod
+    def default(cls) -> Any:
+        return None
+
+    def modify(self, definition: dict[str, Any]) -> None:
+        if self.value is not None:
+            # Create a configmap with system parameters
+            configmap_name = "system-params-test"
+
+            # First create the configmap in the cluster
+            configmap_yaml = f"""
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {configmap_name}
+  namespace: materialize-environment
+data:
+  system-params.json: |
+    {json.dumps(self.value)}
+"""
```

Comment:
Doing string interpolation like this in yaml is unsafe.

Untested suggestion:
```suggestion
            configmap_yaml = yaml.safe_dumps({
                "apiVersion": "v1",
                "kind": "ConfigMap",
                "metadata": {
                    "name": configmap_name,
                    "namespace": "materialize-environment",
                },
                "data": {
                    "system-params.json": json.dumps(self.value),
                },
            })
```